### PR TITLE
[FEATURE] Added enable highlight flag

### PR DESCRIPTION
**Pull request by Deepak-Kesavan**
## What

- Added enable highlight flag

Note: Tool needs to be built once PR is merged.

## Why

- To reduce tokens consumed by highlight if highlighting is not used.

## How

-

## Can this PR break any existing features. If yes, please list possible items. If no, please explain why. (PS: Admins do not merge the PR without this section filled)

-

## Database Migrations

- 

## Env Config

- 

## Relevant Docs

-

## Related Issues or PRs

-

## Dependencies Versions

-

## Notes on Testing

-

## Screenshots

image

## Checklist

I have read and understood the [Contribution Guidelines]().

### CODE REVIEW

File: backend/prompt_studio/prompt_studio_registry/prompt_studio_registry_helper.py [hunk 1]
```diff
@@ -71,6 +71,12 @@ def frame_spec(tool: CustomTool) -> Spec:
                 "default": False,
                 "description": "Enables single pass extraction",
             },
+            "enable_highlight": {
+                "type": "boolean",
+                "title": "Enable highlight",
+                "default": False,
+                "description": "Enables highlight",
```
Comment (chandrasekharan-zipstack) [inline backend/prompt_studio/prompt_studio_registry/prompt_studio_registry_helper.py:78]: NIT: @Deepak-Kesavan elaborate on the description - how will this affect a non MRQ user?

File: prompt-service/src/unstract/prompt_service/helper.py [hunk 2]
```diff
@@ -260,7 +260,9 @@ def construct_and_run_prompt(
     metadata: dict[str, Any],
 ) -> str:
     platform_postamble = tool_settings.get(PSKeys.PLATFORM_POSTAMBLE, "")
-    if tool_settings.get(PSKeys.SUMMARIZE_AS_SOURCE):
+    summarize_as_source = tool_settings.get(PSKeys.SUMMARIZE_AS_SOURCE)
+    enable_highlight = tool_settings.get(PSKeys.ENABLE_HIGHLIGHT, False)
+    if not enable_highlight or summarize_as_source:
```
Comment (chandrasekharan-zipstack) [inline prompt-service/src/unstract/prompt_service/helper.py:265]: @Deepak-Kesavan why is this `not  enable_highlight` added?

File: frontend/src/components/custom-tools/settings-modal/SettingsModal.jsx [hunk 3]
```diff
@@ -95,12 +98,29 @@ function SettingsModal({ open, setOpen, handleUpdateTool }) {
...
+        <HighlightManager
+          handleUpdateTool={handleUpdateTool}
+          type="highlight"
+        />
       );
     }
-
     setMenuItems(items);
     setComponents(listOfComponents);
+    console.log(listOfComponents);
```
Comment (chandrasekharan-zipstack) [inline frontend/src/components/custom-tools/settings-modal/SettingsModal.jsx:123]: NIT: Is this required?

**Review by chandrasekharan-zipstack — APPROVED** [review]

File: backend/sample.env [hunk 4]
```diff
@@ -86,9 +86,9 @@ PROMPT_STUDIO_FILE_PATH=/app/prompt-studio-data
 
 # Structure Tool Image (Runs prompt studio exported tools)
 # https://hub.docker.com/r/unstract/tool-structure
-STRUCTURE_TOOL_IMAGE_URL="docker:unstract/tool-structure:0.0.42"
+STRUCTURE_TOOL_IMAGE_URL="docker:unstract/tool-structure:0.0.43"
```
Comment (chandrasekharan-zipstack) [inline backend/sample.env:89]: @Deepak-Kesavan was this version built? I don't see [this built recently](https://github.com/Zipstack/unstract/actions/workflows/docker-tools-build-push.yaml)

Comment (Deepak-Kesavan) [reply backend/sample.env:89]: I usually build from main only.

File: frontend/src/components/custom-tools/settings-modal/SettingsModal.jsx (hunk 3 above)
Comment (Deepak-Kesavan) [reply frontend/src/components/custom-tools/settings-modal/SettingsModal.jsx:123]: Not required. Will remove it.

File: prompt-service/src/unstract/prompt_service/helper.py (hunk 2 above)
Comment (Deepak-Kesavan) [reply prompt-service/src/unstract/prompt_service/helper.py:265]: No need to extract line numbers if highlight is disabled. That is why it is added.

**Review by vishnuszipstack — APPROVED** [review]

File: backend/prompt_studio/prompt_studio_registry/prompt_studio_registry_helper.py (hunk 1 above)
Comment (Deepak-Kesavan) [reply backend/prompt_studio/prompt_studio_registry/prompt_studio_registry_helper.py:78]: @chandrasekharan-zipstack Let it be like this for now. I am planning to enhance this after checking with @vishnuszipstack and remove this setting itself. We can internally handle this based on if MRQ is selected.

**Pull request merged**